be/src/util/fixed_hash_map.h

```diff
@@ -0,0 +1,125 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#pragma once
+
+#include <cmath>
+#include <cstdint>
+#include <cstring>
+#include <limits>
+#include <optional>
+#include <type_traits>
+#include <utility>
+
+#include "glog/logging.h"
+namespace starrocks::vectorized {
+
+// FixedSizeHashMap
+// Key: KeyType integer type eg: uint8 uint16
+// value shouldn't be nullptr
+template <typename KeyType, typename ValueType>
+class SmallFixedSizeHashMap {
+public:
+    static_assert(std::is_integral_v<KeyType>);
+    static_assert(std::is_pointer_v<ValueType>);
+    static constexpr int hash_table_size = 1 << sizeof(KeyType) * 8;
+    class iterator;
+
+    using key_type = KeyType;
+    using search_key_type = typename std::make_unsigned<KeyType>::type;
+    using iterator = iterator;
```

Comment:
iterator = iterator ?

Reply:
pay attention to the check result tscancode

Reply:
fixed